docusaurus/docs/quickstarts/network/hosted.md

```diff
@@ -1,168 +1,235 @@
+---
+sidebar_position: 10
+---
+# Host OpenZiti Anywhere
+
+You can absolutely choose to host your [OpenZiti Network](../../introduction/01-Introduction.mdx#overview-of-a-ziti-network) anywhere you like.
+It is not necessary for the server to be on the open internet. If it works better for you to deploy OpenZiti on your
+own network, great, do that.  The only requirement to be aware of is that every piece of the a network will need to be able to communicate to the controller at least one edge router.
+
+If you have a server available on the open internet, or you will provision one for use with OpenZiti, that's the
+ideal scenario. With a zero trust overlay network provided by OpenZiti, you can rest assured that your traffic is safe even when using commodity internet. Furthermore, you do not need to worry about being on a network you trust, as all networks are considered untrustworthy, even your work/home network!
+
+## Installation
+
+When starting out deploying an [OpenZiti Network](../../introduction/01-Introduction.mdx#overview-of-a-ziti-network), we recommend you follow
+and use the `expressInstall` function provided by the OpenZiti project. Once you're familiar with the network and
+the configuration options available you'll be better equipped to make changes.
+
+### Firewall
+
+The first issue you will need to deal with is opening some ports. A network will consist of at least one controller and 
+at least one edge router. Both of these components will require ports to be open. For the controller you will need to 
+open a range of ports through your firewall:
+
+- `8440/tcp`: Edge Controller providing router control plane
+- `8441/tcp`: Edge Controller providing client sessions
+- `8442/tcp`: Edge Router providing client connections
+- `8443/tcp`: Ziti Admin Console (ZAC) [optional]
+
+These are the arbitrary ports we'll use in this example for convenience when specifying the firewall exception as a port range.
+
+## Express Install
+
+With the firewall ports open you will now be able to source the script which provides the expressInstall function and
+execute it. You probably should consider DNS names before actually running the script. By default, the script will use the output of the
+`hostname -s` command to determine a reasonable default for the PKI that will be generated as well as the
+configuration files that are output.  The script allows you to configure these values and if you are deploying to a cloud
+provider (AWS, Azure, OCI, IBM, Digital Ocean, GCP, etc.) you will almost certainly want to use the **public** DNS name
+of your instance. You possibly  want to override the IP address to use as well, but using DNS names is superior to using
+IP addresses in case your IP changes.
+
+The quickest and easiest thing to do, is simply find your external DNS name and set it into the EXTERNAL_DNS environment
+variable. For example:
+
+```bash
+export EXTERNAL_DNS="acme.example.com"
```

Comment:
@gberl002 what happens when there IS no external dns? i think you recently changed these words? or are you in that procecss? (i don't recall)

Reply:
I've gotten into this habit when I intend to run the hosted quickstart but don't plan to expose it externally.

```bash
export EXTERNAL_DNS="localhost"
```

That way, I don't need to modify the provided shell env snippet which currently needs `EXTERNAL_DNS`. We could provide a default value for `EXTERNAL_DNS` in that snippet. I'll add that in this branch for review.

Reply:
I think this means we should default `EXTERNAL_DNS` to empty string `""`, not `localhost`, in this hosted quickstart's `expressInstall` setup section. For example:

```bash
export EXTERNAL_DNS=${EXTERNAL_DNS:-""}
export EXTERNAL_IP="$(curl -s eth0.me)"       
export ZITI_EDGE_CONTROLLER_IP_OVERRIDE="${EXTERNAL_IP}"
export ZITI_EDGE_ROUTER_IP_OVERRIDE="${EXTERNAL_IP}"
export ZITI_EDGE_CONTROLLER_HOSTNAME="${EXTERNAL_DNS}"
export ZITI_EDGE_ROUTER_HOSTNAME="${EXTERNAL_DNS}"
export ZITI_CTRL_PORT=8440
export ZITI_EDGE_CONTROLLER_PORT=8441
export ZITI_EDGE_ROUTER_PORT=8442
```

That way, if they don't set `EXTERNAL_DNS` it'll default to `ZITI_EDGE_CTRL_ADVERTISED_HOST_PORT`.

Reply:
Actually, an unset var is practically equivalent to an empty string value in this particular case because we're not catching errors (`errexit`) or unset vars (`nounset`) because the shell is presumed interactive. So, we don't really need to add the line to explicitly set `EXTERNAL_DNS`. If they skip the setup step to assign `EXTERNAL_DNS` then it'll just work.